src/v/utils/tests/tracking_allocator_tests.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Redpanda Data, Inc.
+//
```

Comment:
I think you should test more scenarios, here, e.g., move and copy constructors, move and copy assignment, since especially there things can go sideways with allocators.

It would be nice to see a description somewhere of e.g. how move constructor works for a container using these allocators: are the counters in the source allocator zeroed out or do they keep their values? How about the destination?